File: mvn-advent-of-code/src/main/java/advent/twenty_fifteen/Day04TheIdealStockingStuffer.java
package advent.twenty_fifteen;

import advent.common.DailyProblem;
import advent.utilities.MessageDigestUtilities;

import java.security.MessageDigest;
import java.security.NoSuchAlgorithmException;

public class Day04TheIdealStockingStuffer implements DailyProblem<Integer, Integer> {

    private final int part1Answer;
    private final int part2Answer;

    public Day04TheIdealStockingStuffer(String key) throws NoSuchAlgorithmException {
        MessageDigestUtilities mdUtilities = new MessageDigestUtilities();

        this.part1Answer = mdUtilities.findNextHashWithXZeros(key, 9999, 5).getSecond();
        this.part2Answer = mdUtilities.findNextHashWithXZeros(key, 9999, 6).getSecond();
    }

    @Override
    public Integer getPart1Answer() {
        return part1Answer;
    }

    @Override
    public Integer getPart2Answer() {
        return part2Answer;
    }
}

File: scala-advent-of-code/src/main/scala/advent/twenty_fifteen/Day04TheIdealStockingStuffer.scala
package advent.twenty_fifteen

import advent.common.DailyProblem
import advent.utilities.ChecksumUtilities

class Day04TheIdealStockingStuffer(key: String) extends DailyProblem[Int, Int] {

  override val part1Answer: Int = ChecksumUtilities.findNextHashWithXZeros(key, 0, 5)._2
  override val part2Answer: Int = ChecksumUtilities.findNextHashWithXZeros(key, 0, 6)._2
}
